serialization.go

```diff
@@ -225,6 +225,18 @@ func MerkleTreeLeafForEmbeddedSCT(chain []*x509.Certificate, timestamp uint64) (
 	}, nil
 }
 
+// LeafHashForLeaf returns the leaf hash for a Merkle tree leaf.
+func LeafHashForLeaf(leaf *MerkleTreeLeaf) ([sha256.Size]byte, error) {
```

Comment:
My only wondering about this PR is whether this function should be put with the other LeafHash function, so that if anyone goes a-looking for leaf hash functions they will all be in the same place to take their pick from, rather than finding one and then assuming there aren't any others when there are?

Happy to be convinced otherwise though :)

Reply:
This one seemed to just use the base CT types (it didn't need anything outside `ct`) so it felt more appropriate here.